nft-barter-core/contracts/nft-barter.sol

```diff
@@ -1,7 +1,212 @@
+            valueDifference,
+            _getNextId(),
+            makerTokenId,
+            takerTokenId
+        );
+        _updateSwapsData(swap);
+
+        emit SwapInitiated(swap);
+
+        return swap;
+    }
+
+    function updateSwapValue(uint128 swapId, int152 valueDifference)
+        external
+        override
+        onlyIfSwapExists(swapId)
+        onlyIfParticipant(swapId)
+        returns (SwapOrder memory)
+    {
+        SwapOrder memory swap = _swaps[swapId];
+        swap.valueDifference = valueDifference;
+
+        _updateSwapsData(swap);
+
+        emit SwapUpdate("ValueUpdate", swap);
+        return swap;
+    }
+
+    function updateSwapMakerToken(uint128 swapId, uint256 makerTokenId)
+        external
+        override
+        onlyIfSwapExists(swapId)
+        onlyIfTokenIsValid(makerTokenId)
+        onlyIfParticipant(swapId)
+        returns (SwapOrder memory)
+    {
+        SwapOrder memory swap = _swaps[swapId];
+        swap.makerTokenId = makerTokenId;
+        _updateSwapsData(swap);
+
+        emit SwapUpdate("MakerUpdate", swap);
+        return swap;
+    }
+
+    function updateSwapTakerToken(uint128 swapId, uint256 takerTokenId)
+        external
+        override
+        onlyIfSwapExists(swapId)
+        onlyIfTokenIsValid(takerTokenId)
+        onlyIfParticipant(swapId)
+        returns (SwapOrder memory)
+    {
+        SwapOrder memory swap = _swaps[swapId];
+        swap.takerTokenId = takerTokenId;
+        _updateSwapsData(swap);
+
+        emit SwapUpdate("MakerUpdate", swap);
+        return swap;
+    }
+
+    function cancelSwap(uint128 swapId)
+        external
+        override
+        onlyIfSwapExists(swapId)
+        onlyIfParticipant(swapId)
+        returns (SwapOrder memory)
+    {
+        SwapOrder memory clearedSwap = _clearSwapsData(swapId);
+        emit SwapCanceled(clearedSwap);
+        return clearedSwap;
+    }
+
+    function acceptSwap(uint128 swapId, uint256 takerTokenId)
+        external
+        payable
+        override
+        onlyIfSwapExists(swapId)
+        onlyIfTokenIsValid(takerTokenId)
+        onlyIfTaker(swapId)
+        returns (bool)
+    {
+        require(isSwapPossible(swapId), USELESS_SWAP);
+        SwapOrder memory swap = _swaps[swapId];
+        require(swap.takerTokenId == takerTokenId, INVALID_TOKEN_ID); //todo: discuss with team wheather to include this or not
+        ERC721._transfer(
+            swap.makerAddress,
+            swap.takerAddress,
+            swap.makerTokenId
+        );
+        ERC721._transfer(
+            swap.takerAddress,
+            swap.makerAddress,
+            swap.takerTokenId
+        );
+        emit SwapAccepted(swap);
+        return true;
+    }
+
+    function isSwapPossible(uint128 swapId)
+        public
+        view
+        onlyIfSwapExists(swapId)
+        returns (bool)
+    {
+        SwapOrder memory swap = _swaps[swapId];
+        address maker = ERC721.ownerOf(swap.makerTokenId);
```

Comment:
we can rewrite this as 
`return (maker != swap.makerAddress) && taker != swap.takerAddress) `

Infact we can remove the variables `maker & taker`